__tests__/app.test.js

```diff
@@ -22,86 +22,200 @@ describe('GET /api', () => {
+            .toBeInstanceOf(Object);
+          expect(body.endpoints['GET /api/articles'].exampleResponse)
+            .toEqual(endpoints['GET /api/articles'].exampleResponse);
+        });
+    });
 });
-describe('/api/articles', () => {
-  test('GET:200 sends articles to the client', () => {
+describe('/api/articles/:article_id/comments', () => {
+  test('GET:200 sends comments by article_id to the client', () => {
     return request(app)
-      .get('/api/articles')
+      .get('/api/articles/1/comments')
       .expect(200)
       .then(({ body }) => {
-        expect(body.articles).toBeInstanceOf(Array);
-        expect(body.articles.length > 0).toBe(true);
-        body.articles.forEach((article) =>{
-          expect(article).toBeInstanceOf(Object);
-          expect(typeof article.article_id).toBe('number');
-          expect(typeof article.title).toBe('string');
-          expect(typeof article.author).toBe('string');
-          expect(typeof article.topic).toBe('string');
-          expect(typeof article.created_at).toBe('string');
-          expect(typeof article.votes).toBe('number');
-          expect(typeof article.article_img_url).toBe('string');
-          expect(typeof article.comment_count).toBe('number');
-        })
+        expect(body.comments).toBeInstanceOf(Array);
+        expect(body.comments.length > 0).toBe(true);
+        body.comments.forEach((comment) => {
+          expect(comment).toBeInstanceOf(Object);
+          expect(typeof comment.comment_id).toBe('number');
+          expect(typeof comment.votes).toBe('number');
+          expect(typeof comment.created_at).toBe('string');
+          expect(typeof comment.author).toBe('string');
+          expect(typeof comment.body).toBe('string');
+          expect(typeof comment.article_id).toBe('number');
+        });
       });
   });
+  test(
+    'GET:200 sends comments to the client sorted by date in descending order',
+    () => {
+      return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({ body }) => {
+          const sortedArray = [...body.comments];
+          sortedArray.sort((a, b) => {
+            return Date.parse(b.created_at) - Date.parse(a.created_at);
+          });
+          expect(body.comments).toEqual(sortedArray);
+        });
+    });
+  test(
+    'GET:400 sends an appropriate status and error message when given an invalid id',
+    () => {
+      return request(app)
+        .get('/api/articles/not-valid-id/comments')
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe('Bad Request');
+        });
+    });
+  test(
+    'GET:404 sends an appropriate status and error message when given a valid but non-existent id',
+    () => {
+      return request(app)
+        .get('/api/articles/999/comments')
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe('Comments does not exist');
+        });
+    });
```

Comment:
Is it that a 'comment does not exist', or is it that the article id does not exist? If an article does not exist then it won't have any comments anyway.